File: src/types/preset.ts
import type { FFmpegCommand } from './ffmpeg';

/** プリセット */
export interface Preset {
  id: string;
  name: string;
  description: string;
  category: PresetCategory;
  isBuiltin: boolean;
  command: FFmpegCommand;
  createdAt?: string;
  updatedAt?: string;
}

export type PresetCategory = 'web' | 'social' | 'archive' | 'audio' | 'custom';

export interface PresetList {
  builtin: Preset[];
  user: Preset[];
}

File: src-tauri/src/models/preset.rs
use super::ffmpeg::FFmpegCommand;
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct Preset {
    pub id: String,
    pub name: String,
    pub description: String,
    pub category: PresetCategory,
    pub is_builtin: bool,
    pub command: FFmpegCommand,
    pub created_at: Option<String>,
    pub updated_at: Option<String>,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
#[serde(rename_all = "snake_case")]
pub enum PresetCategory {
    Web,
    Social,
    Archive,
    Audio,
    Custom,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct PresetList {
    pub builtin: Vec<Preset>,
    pub user: Vec<Preset>,
}
